akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala

```diff
@@ -40,56 +40,78 @@ trait ActorContext[T] {
 
   /**
    * Get the `scaladsl` of this `ActorContext`.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as [[java.util.concurrent.CompletionStage]] callbacks.
    */
   def asScala: akka.actor.typed.scaladsl.ActorContext[T]
 
   /**
    * The identity of this Actor, bound to the lifecycle of this Actor instance.
    * An Actor with the same name that lives before or after this instance will
    * have a different [[ActorRef]].
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as [[java.util.concurrent.CompletionStage]] callbacks.
    */
   def getSelf: ActorRef[T]
 
-  /**
-   * Return the mailbox capacity that was configured by the parent for this actor.
-   */
-  def getMailboxCapacity: Int
-
   /**
    * The [[ActorSystem]] to which this Actor belongs.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as [[java.util.concurrent.CompletionStage]] callbacks.
    */
   def getSystem: ActorSystem[Void]
 
   /**
    * The list of child Actors created by this Actor during its lifetime that
    * are still alive, in no particular order.
+   *
+   * *Warning*: This method is not thread-safe and must not be accessed from other threads
+   * than the ordinary actor message processing thread, such as [[java.util.concurrent.CompletionStage]] callbacks.
```

Comment:
Might be worth noting that the result can be shared with other threads if published safely as it is a copy

Reply:
It actually is thread safe in both ways, I changed this to the somewhat incorrect warning to discourage users from doing any kind of fishing for children from another thread as that will be stale information and general bad practice, after getting feedback from Patrik about it.